app/models/user.rb

```diff
@@ -1963,6 +1967,28 @@ def set_default_sidebar_section_links
     SidebarSectionLink.insert_all(records) if records.present?
   end
 
+  def update_default_sidebar_section_links
+    return if !SiteSetting.enable_experimental_sidebar_hamburger
+    return if staged? || bot?
+
+    if SiteSetting.default_sidebar_categories.present?
+      default_category_ids = SiteSetting.default_sidebar_categories.split("|")
+
+      # Filters out categories that user does not have access to or do not exist anymore
+      category_ids = Category.secured(self.guardian).where(id: default_category_ids).pluck(:id)
+      existing_category_ids = SidebarSectionLink.where(user: self, linkable_type: 'Category').pluck(:linkable_id)
+
+      # Only include secured categories
+      secured_category_ids = self.secure_category_ids
+
+      # Secured categories that are in the default categories list
+      categories_to_update = existing_category_ids + (category_ids & secured_category_ids)
+
+      updater = UserUpdater.new(self, self)
+      updater.update({ sidebar_category_ids: [categories_to_update] })
+    end
+  end
```

Comment:
It feels like we're partially re-implementing the addition method here with a significant amount of logic being duplicated. Will we be able to refactor both methods such that the amount of duplication is reduced?